src/components/MDXComponents/MDXHeading.tsx

```diff
@@ -4,16 +4,19 @@ import slug from './utils/slug';
 
 export const MDXHeading = (props) => {
   const { level, children } = props;
-
   let href = '';
 
   if (children && typeof children != 'string') {
     let newChildren = '';
-    for (let i = 0; i < children.length; i++) {
-      if (typeof children[i] == 'string') {
-        newChildren = newChildren + children[i];
-      } else if (typeof children[i] != 'string') {
-        newChildren = newChildren + children[i].props.children;
+    if (typeof children == 'object' && children.props?.children) {
+      newChildren = children.props.children;
+    } else {
+      for (let i = 0; i < children.length; i++) {
+        if (typeof children[i] == 'string') {
+          newChildren = newChildren + children[i];
+        } else {
+          newChildren = newChildren + children[i].props.children;
+        }
```

Comment:
not blocking: Would it be okay to add comments around these if/else branches to explain what cases they are?

Something similar to what you had in the description where we're checking the typeof `children` is an `object` because it represents a single `<code>` (or other HTML) element. Similarly, we have the else statement because the mdx could be an array of elements and that covers the case where we have strings and some sort of HTML element (like `<code>`) right?

Reply:
Yeah good point, I had to do some double checking what they were for also, so I will add some comments. And to your question, yes, that's correct.